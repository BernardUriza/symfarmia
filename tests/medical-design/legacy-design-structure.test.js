const fs = require('fs');
const path = require('path');

describe('Legacy design directory structure', () => {
  const baseDir = path.resolve(__dirname, '../../legacy-design');

  it('should have design-tokens directory', () => {
    expect(fs.existsSync(path.join(baseDir, 'design-tokens'))).toBe(true);
  });

  it('should have component-registry directory', () => {
    expect(fs.existsSync(path.join(baseDir, 'component-registry'))).toBe(true);
  });

  it('should have figma-exports directory', () => {
    expect(fs.existsSync(path.join(baseDir, 'figma-exports'))).toBe(true);
  });

  it('should have current symlink in figma-exports', () => {
    expect(fs.existsSync(path.join(baseDir, 'figma-exports', 'current'))).toBe(true);
  });

  it('should have legacy_design_system_docs.md file', () => {
    expect(fs.existsSync(path.join(baseDir, 'legacy_design_system_docs.md'))).toBe(true);
  });

  it('should have ComponentMigrationTemplate.md file', () => {
    expect(fs.existsSync(path.join(baseDir, 'ComponentMigrationTemplate.md'))).toBe(true);
  });

  it('should have migration-log.md file', () => {
    expect(fs.existsSync(path.join(baseDir, 'migration-log.md'))).toBe(true);
  });

  it('should have scripts directory', () => {
    expect(fs.existsSync(path.join(baseDir, 'scripts'))).toBe(true);
  });
});