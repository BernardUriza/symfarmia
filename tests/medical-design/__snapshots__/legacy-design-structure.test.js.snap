// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legacy design directory structure should match the approved directory structure 1`] = `
[
  "ComponentMigrationTemplate.md",
  "README.md",
  {
    "component-registry": [
      "alerts.md",
      "buttons.md",
      "cards.md",
      "forms.md",
      "medical-components.md",
    ],
  },
  {
    "design-tokens": [
      "colors.json",
      "medical-theme.json",
      "spacing.json",
      "typography.json",
    ],
  },
  {
    "figma-exports": [
      {
        "2025-01-15-medical-v1": [
          "App.tsx",
          "Attributions.md",
          "Guidelines.md",
          {
            "components": [
              "ClinicalNotes.tsx",
              "ConversationCapture.tsx",
              "DialogueFlow.tsx",
              "OrderEntry.tsx",
              "SummaryExport.tsx",
              "ThemeProvider.tsx",
              {
                "figma": [
                  "ImageWithFallback.tsx",
                ],
              },
              {
                "ui": [
                  "accordion.tsx",
                  "alert-dialog.tsx",
                  "alert.tsx",
                  "aspect-ratio.tsx",
                  "avatar.tsx",
                  "badge.tsx",
                  "breadcrumb.tsx",
                  "button.tsx",
                  "calendar.tsx",
                  "card.tsx",
                  "carousel.tsx",
                  "chart.tsx",
                  "checkbox.tsx",
                  "collapsible.tsx",
                  "command.tsx",
                  "context-menu.tsx",
                  "dialog.tsx",
                  "drawer.tsx",
                  "dropdown-menu.tsx",
                  "form.tsx",
                  "hover-card.tsx",
                  "input-otp.tsx",
                  "input.tsx",
                  "label.tsx",
                  "menubar.tsx",
                  "navigation-menu.tsx",
                  "pagination.tsx",
                  "popover.tsx",
                  "progress.tsx",
                  "radio-group.tsx",
                  "resizable.tsx",
                  "scroll-area.tsx",
                  "select.tsx",
                  "separator.tsx",
                  "sheet.tsx",
                  "sidebar.tsx",
                  "skeleton.tsx",
                  "slider.tsx",
                  "sonner.tsx",
                  "switch.tsx",
                  "table.tsx",
                  "tabs.tsx",
                  "textarea.tsx",
                  "toggle-group.tsx",
                  "toggle.tsx",
                  "tooltip.tsx",
                  "use-mobile.ts",
                  "utils.ts",
                ],
              },
            ],
          },
          {
            "styles": [
              "globals.css",
            ],
          },
        ],
      },
      {
        "current": "-> 2025-01-15-medical-v1
",
      },
    ],
  },
  "legacy_design_system_docs.md",
  "migration-log.md",
]
`;
